src/System.cpp

```diff
@@ -249,6 +244,9 @@ void System::toggleFullscreen() {
     SDL_GetDesktopDisplayMode(0, &desktopMode);
     SDL_SetWindowPosition(_window, 10, 10);
     SDL_SetWindowSize(_window, desktopMode.w, desktopMode.h);
+    // We force a reshape even though that should happen as a consequence of SDL window resize events.
+    // But that seems to work unreliably on some OS's, hence we explicitly force it.
+    Control::instance().reshape(desktopMode.w, desktopMode.h);
```

Comment:
This probably causes multiple consecutive reshapes on platforms where this line is not necessary but it seems to work on those anyway.